### PR TITLE
Return default start value for enumerations

### PR DESCRIPTION
**Pull request by hkiel**
Fix missing handling of enumerations as requested in #8558

### CODE REVIEW

File: testsuite/simulation/modelica/initialization/parameters.mos [hunk 1]
```diff
@@ -48,35 +53,45 @@ simulate(initializationTests.parameters, simflags="-lv=LOG_INIT_V"); getErrorStr
 // |                 | |       | | | [2] parameter Real r2(start=2, fixed=false) = 2
 // |                 | |       | | | [3] parameter Real r3(start=3, fixed=true) = 3
 // |                 | |       | | integer parameters
-// |                 | |       | | | [1] parameter Integer i1(start=0, fixed=true) = 0
-// |                 | |       | | | [2] parameter Integer i2(start=2, fixed=false) = 2
-// |                 | |       | | | [3] parameter Integer i3(start=3, fixed=true) = 3
+// |                 | |       | | | [1] parameter Integer e1(start=0, fixed=true) = 0
```
Comment (hkiel) [inline testsuite/simulation/modelica/initialization/parameters.mos:56]: enums are listed under `integer parameters`, is that correct?

**Comment by hkiel** [comment]
Has anyone got an idea on how to fix this? `testsuite/simulation/modelica/initialization/parameters.mos`
```
assert | warning | The following assertion has been violated at time 0.000000
| | |       | e1 >= initializationTests.E.E1 and e1 <= initializationTests.E.En
assert | warning | Variable violating min/max constraint: initializationTests.E.E1 <= e1 <= initializationTests.E.En, has value: 0
```
I thought this should be fine, because `initializationTests.E.E1` is taken as default!?
```
[<interactive>:10:5-10:19:writable] Warning: Parameter e1 has no value, and is fixed during initialization (fixed=true), using available start value (start=initializationTests.E.E1) as default value.
```

File: OMCompiler/Compiler/FrontEnd/DAEUtil.mo [hunk 2]
```diff
@@ -903,20 +903,24 @@ end setMinMax;
...
+    case (SOME(DAE.VAR_ATTR_REAL(start = NONE())), _) then DAE.RCONST(0.0);
+    case (SOME(DAE.VAR_ATTR_INT(start = SOME(r))), _) then r;
+    case (SOME(DAE.VAR_ATTR_INT(start = NONE())), _) then DAE.ICONST(0);
+    case (SOME(DAE.VAR_ATTR_BOOL(start = SOME(r))), _) then r;
+    case (SOME(DAE.VAR_ATTR_BOOL(start = NONE())), _) then DAE.BCONST(false);
+    case (SOME(DAE.VAR_ATTR_STRING(start = SOME(r))), _) then r;
+    case (SOME(DAE.VAR_ATTR_STRING(start = NONE())), _) then DAE.SCONST("");
+    case (SOME(DAE.VAR_ATTR_ENUMERATION(start = SOME(r))), _) then r;
+    case (SOME(DAE.VAR_ATTR_ENUMERATION(start = NONE())), DAE.T_ENUMERATION(path = path, names = names)) then DAE.ENUM_LITERAL(AbsynUtil.joinPaths(path, Absyn.IDENT(listHead(names))), 1);
     else DAE.RCONST(0.0);
```
Comment (lochel) [inline OMCompiler/Compiler/FrontEnd/DAEUtil.mo:924]: Should we remove the else case?

**Comment by hkiel** [comment]
> ```
> assert | warning | The following assertion has been violated at time 0.000000
> | | |       | e1 >= initializationTests.E.E1 and e1 <= initializationTests.E.En
> assert | warning | Variable violating min/max constraint: initializationTests.E.E1 <= e1 <= initializationTests.E.En, has value: 0
> ```

This warning also appears without my changes to DAEUtil when adding the `parameter E e1;` line to the test.

I'll add the expected output and remove the `else` case as suggested by @lochel

**Review by lochel — APPROVED** [review]
Looks good. Thanks for your help.

**Comment by hkiel** [comment]
Hum, ... the latest run with only one modified test case, now produces tons of errors in testsuite.
Looks like something unpredictable happens (garbage collection?).

On my machine the two statemachine tests keep failing (`testsuite/simulation/modelica/statemachines/Ticket4618.mos` and `testsuite/simulation/modelica/statemachines/EnteringByReset.mos`)

```
Equation mismatch: diff says:
--- /tmp/omc-rtest-henning/simulation/modelica/statemachines/equations-expected	2022-02-17 12:49:33.000000000 +0100
+++ /tmp/omc-rtest-henning/simulation/modelica/statemachines/equations-got	2022-02-17 12:49:34.000000000 +0100
@@ -1,4 +1,4 @@
 true
 true
-0.7
-0.4
+
+
```

I tried using the original getStartAttr() function in `OMCompiler/Compiler/FrontEnd/StateMachineFlatten.mo`, but that does not help either.

**Comment by perost** [comment]
Removing the `else` means the function will fail on `DAE.VAR_ATTR_CLOCK`, so that's probably why a lot of tests fail now.

**Comment by hkiel** [comment]
testsuite works again with the last commit.
Looks like enumerations without start value are handled best as RCONST(0.0) instead of enumeration item.

Maybe that could be a problem with the enum item not being considered constant?
As far as I can see the failing tests do not have enums, but Clock variables, but these are not changed.

@adrpo @sjoelund @lochel

**Comment by hkiel** [comment]
@lochel Lennart, can you merge this PR, please? All troubles are sorted out now. I squashed all those commits into one.

**Comment by lochel** [comment]
Thanks Henning!

**Pull request merged**